src/EditorFeatures/Core/Shared/Utilities/CaretPreservingEditTransaction.cs

```diff
@@ -30,6 +30,20 @@ internal class CaretPreservingEditTransaction : IDisposable
             }
         }
 
+        public static CaretPreservingEditTransaction TryCreate(string description, 
+            ITextView textView,
+            ITextUndoHistoryRegistry undoHistoryRegistry,
+            IEditorOperationsFactoryService editorOperationsFactoryService)
+        {
+            ITextUndoHistory unused;
+            if (undoHistoryRegistry.TryGetHistory(textView.TextBuffer, out unused))
```

Comment:
You could add a new private constructor that used this and have this call that instead.